src/commands/login.rs

```diff
@@ -367,6 +372,18 @@ impl LoginCommand {
         }
     }
 
+    fn validate_hippo_url(&self) -> Result<(), anyhow::Error> {
```

Comment:
I would say this should probably happen in `fn url(&self)`, i.e. changing its signature to return `Result<Url>`

Reply:
We would then need to randomly call `self.url()?;` to validate early which feels a _bit_ weird conceptually, but otherwise seems fine to me.

Reply:
Another option would be to let `clap` (the CLI arg parser) validate this. That is super easy if you don't care about precise error messaging but a little trickier if you want to customize errors (as in I don't remember how to do it off the top of my head :sweat_smile:).

Reply:
nit: I think we now call the flag just `--url` — could be `validate_url` (or something similar) instead?